lib/improver/argparser.py

```diff
@@ -28,21 +28,190 @@
+              {'action': 'store_true', 'default': False, 'help': ... })
+
+    The CENTRALIZED_ARGUMENTS will be selected from, as necessary, for each
+    of the CLIs that we create, and the COMPULSORY_ARGUMENTS will be
+    automatically added to all CLIs (with no option to exclude them).
+    """
+
+    # Ideally, all CLIs should select something from this dictionary:
+    # NB: --help included by default with ArgumentParser
+    CENTRALIZED_ARGUMENTS = {
+        'input_file': (
+            ['input_filepath'],
+            {'metavar': 'INPUT_FILE',
+             'help': 'A path to an input NetCDF file to be processed'}),
+        'output_file': (
+            ['output_filepath'],
+            {'metavar': 'OUTPUT_FILE',
+             'help': 'The output path for the processed NetCDF'}),
+    }
+
+    # *All* CLIs will use the options here (no option to disable them):
+    COMPULSORY_ARGUMENTS = {
+        # TODO: Implement (some of?) these:
+        #        'debug': (
+        #            ['--debug'],
+        #            {'action': 'store_true'}),
+        #        'dry_run': (
+        #            ['--dry-run'],
+        #            {'action': 'store_true'}),
+        #        'profile' : (
+        #            ['--profile'],
+        #            {'action': 'store_true'}),
+        #        'verbose': (
+        #            ['--verbose', '-v'],
+        #            {'action': 'count'}),
+        #        'version': (
+        #            ['--version'],
+        #            {'action': 'version',
+        #             'version': improver.version_message() }),
+    }
+
+    # We can override including these, but options common to everything should
+    # be in a list here:
+    # DEFAULT_CENTRALIZED_ARG_NAMES = ('input_file', 'output_file')
+    DEFAULT_CENTRALIZED_ARG_NAMES = ()
+
+    def __init__(self, central_arguments=DEFAULT_CENTRALIZED_ARG_NAMES,
+                 specific_arguments=None, **kwargs):
+        """Create an ArgParse instance, which is a subclass of
+        argparse.ArgumentParser and automatically add all of the arguments.
+        (Note: The ArgParse.COMPULSORY_ARGUMENTS are always added.)
+
+        Args:
+            central_arguments (list):
+              A list containing the centralized arguments we require.
+              (Keys of the centralized argument dictionary). By default this is
+              set as ArgParse.DEFAULT_CENTRALIZED_ARG_NAMES.
+            specific_arguments (list):
+              A list of argument specifications required to add arguments
+              which are not contained within the centralized argument
+              dictionary. The format of these argument specifications should
+              be the same as the values in the ArgParser.CENTRALIZED_ARGUMENTS
+              dictionary.
+              (For more details, see the add_arguments method).
+              Default is None, which does not add additional arguments.
+            kwargs (dictionary):
```

Comment:
It might be useful just to say something like "For example the description of the ArgumentParser" as this is the one we use most often and people may want to know how to pass this in. What do you think?